packages/web/src/components/search/SearchBox.js

```diff
@@ -0,0 +1,1230 @@
+				finalQuery = {
+					query_string: shouldQuery(value, fields, props),
+				};
+			} else if (props.searchOperators) {
+				finalQuery = {
+					simple_query_string: shouldQuery(value, fields, props),
+				};
+			} else {
+				finalQuery = {
+					bool: {
+						should: shouldQuery(value, fields, props),
+						minimum_should_match: '1',
+					},
+				};
+			}
+		}
+
+		if (value === '') {
+			finalQuery = {
+				bool: {
+					should: shouldQuery(value, fields, props),
+					minimum_should_match: '1',
+				},
+			};
+		}
+
+		if (finalQuery && props.nestedField) {
+			finalQuery = {
+				nested: {
+					path: props.nestedField,
+					query: finalQuery,
+				},
+			};
+		}
+		return finalQuery;
+	};
+
+	const triggerDefaultQuery = (paramValue) => {
+		const value = typeof paramValue !== 'string' ? currentValue : paramValue;
+		let query = searchBoxDefaultQuery(value, props);
+		if (defaultQuery) {
+			const defaultQueryTobeSet = defaultQuery(value, props) || {};
+			if (defaultQueryTobeSet.query) {
+				({ query } = defaultQueryTobeSet);
+			}
+			// Update calculated default query in store
+			updateDefaultQuery(componentId, props, value);
+		}
+		props.updateQuery({
+			componentId: internalComponent,
+			query,
+			value,
+			componentType: componentTypes.searchBox,
+		});
+	};
+
+	const triggerCustomQuery = (paramValue, categoryValue = undefined) => {
+		const value = typeof paramValue !== 'string' ? currentValue : paramValue;
+		let query = searchBoxDefaultQuery(value, props);
+		if (customQuery) {
+			const customQueryTobeSet = customQuery(value, props) || {};
+			const queryTobeSet = customQueryTobeSet.query;
+			if (queryTobeSet) {
+				query = queryTobeSet;
+			}
+			updateCustomQuery(componentId, props, value);
+		}
+		props.updateQuery({
+			componentId,
+			value,
+			query,
+			label: filterLabel,
+			showFilter,
+			URLParams,
+			componentType: componentTypes.searchBox,
+			category: categoryValue,
+		});
+	};
+
+	const triggerQuery = ({
+		isOpen = false,
+		customQuery = false,
+		defaultQuery = false,
+		value = undefined,
+		categoryValue = undefined,
+	}) => {
+		if (isOpen) {
+			setIsOpen(isOpen);
+		}
+
+		if (customQuery) {
+			triggerCustomQuery(value, categoryValue);
+		}
+		if (defaultQuery) {
+			triggerDefaultQuery(value);
+		}
+	};
+	const withTriggerQuery = (func) => {
+		if (func) {
+			return e => func(e, triggerQuery);
+		}
+		return undefined;
+	};
+	const onValueSelected = (valueSelected = currentValue, ...cause) => {
+		const { onValueSelected } = props;
+		if (onValueSelected) {
+			onValueSelected(valueSelected, ...cause);
+		}
+	};
+	const handleTextChange = debounce((valueParam = undefined, cause = undefined) => {
+		if (cause === causes.CLEAR_VALUE) {
+			triggerCustomQuery(valueParam);
+			triggerDefaultQuery(valueParam);
+		} else if (props.autosuggest) {
+			triggerDefaultQuery(valueParam);
+		} else {
+			triggerCustomQuery(valueParam);
+		}
+	}, props.debounce);
+
+	function setValue(
```

Comment:
@mohdashraf010897 Can you use the same pattern (using `const`) to define the functions?

Reply:
@bietkul fixed here https://github.com/appbaseio/reactivesearch/pull/1806/commits/0a8aff3dfb6c6cf9c72ff92b1c72c7d037d2c3ea